### PR TITLE
Update mocha related dependencies

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Use latest dependencies. 

**Description**

- Update Typescript 
- Update Mocha 
- Update Chai
- Update Yargs

**Steps to test or reproduce**

- Run all tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,22 +25,22 @@
...
-    "@types/mocha": "^8.2.2",
+    "@types/mocha": "^10.0.9",
     "@types/node": "^15.12.4",
     "@types/rimraf": "^3.0.0",
-    "@types/yargs": "^17.0.0",
+    "@types/yargs": "^17.0.33",
     "@typescript-eslint/eslint-plugin": "^4.28.0",
     "@typescript-eslint/parser": "^4.28.0",
-    "chai": "^4.3.4",
+    "chai": "^4.5.0",
```
Comment (nazarhussain) [inline package.json:34]: That's not latest version, but last version supporting commonjs.

**Comment by dapplion** [comment]
Thanks! There's CI failures

**Comment by nazarhussain** [comment]
@dapplion  Seems like the error related tot he permission for the repo. 

> HttpError: Resource not accessible by integration

As locally it's passing fine. 

```
❯ yarn benchmark --local ./cache
yarn run v1.22.22
$ node -r ts-node/register src/cli.ts 'test/perf/**/*.test.ts' --local ./cache
(node:82834) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Connected to historyProvider: LocalHistoryProvider, dirpath ./cache
No previous bencharmk found for latestCommitInBranch 'main'


  Array iteration
    ✔ Regular test
    ✔ sum array with raw for loop                                         849.7938 ops/s    1.176756 ms/op        -       8683 runs   10.7 s
    ✔ sum array with reduce                                               170.7566 ops/s    5.856289 ms/op        -        127 runs   1.25 s
    ✔ sum array with reduce beforeEach                                    171.8360 ops/s    5.819502 ms/op        -        171 runs   1.56 s
    ✔ sum array with reduce before beforeEach                             175.4948 ops/s    5.698174 ms/op        -        220 runs   1.76 s
    - sum array with reduce
    ✔ sum array with reduce high threshold                                175.2783 ops/s    5.705213 ms/op        -        548 runs   3.63 s
    ✔ sum array with reduce no threshold                                  180.2968 ops/s    5.546410 ms/op        -        152 runs   1.35 s


  7 passing (20s)
  1 pending

✨  Done in 21.71s.
```

**Comment by dapplion** [comment]
Would like to have green CI, do you have capacity to look into the permissions issue? I can make you mantainer too

**Comment by nazarhussain** [comment]
> Would like to have green CI, do you have capacity to look into the permissions issue? I can make you mantainer too

Sure that will be helpful to debug.

**Comment by nazarhussain** [comment]
Closing in favor of https://github.com/dapplion/benchmark/pull/24

**Pull request closed without merging**